### ISSUE TITLE
Add loadJuicyWithMetadata function

### ISSUE THREAD

**Issue by ad-si** [alpmestan/gloss-juicy#12]
This should wrap Juicy's `readImageWithMetadata` in the same way as `loadJuicy` wraps `readImage`.

@hpacheco I couldn't create an issue in your fork, so I'm doing it here.

**Comment by georgefst** [comment]
Yep, this would be great. I really need height and width data, so I'm unable to use this library at all, seeing as its functions only ever return `Picture`. The easiest thing for me to do is go via `ByteString` which is suboptimal (then again, this library appears to use `promoteImage` everywhere, which performs a full copy...).

Anyway, I'm not confident about the maintenance status of this library. Issues being disabled is always a bad sign, plus @hpacheco hasn't been active in a while.

**Comment by alpmestan** [comment]
I have to admit I haven't used this library in a long time, but could definitely give commit + hackage upload rights to whoever wants to keep this ball rolling. @ad-si @georgefst Interested?

**Comment by georgefst** [comment]
I'll give it a few weeks to see if @hpacheco replies, and to get a sense of whether I'm likely to use this package long-term. But, yeah, I might be interested.

**Comment by alpmestan** [comment]
Sounds good, just ping me when/if needed.

**Comment by hpacheco** [comment]
Hi guys, like @alpmestan I haven't used gloss and this package in a while. I would be happy to pass the maintainer to someone who has been actively using the package. 
@ad-si is [this](https://github.com/hpacheco/gloss-juicy/commit/acbdfd864b7e574dfbe1595fbdcd154563b2a0ef) what you had in mind? 
P.S. You may open issues in this repo.

**Comment by ad-si** [comment]
> @ad-si is [this](https://github.com/hpacheco/gloss-juicy/commit/acbdfd864b7e574dfbe1595fbdcd154563b2a0ef) what you had in mind?

Perfect, thanks! 😊
[Integrated in Perspec](https://github.com/feramhq/Perspec/commit/6df2e8964380b7d579347271c747b29bb11ab0bc)

> commit + hackage upload rights to whoever wants to keep this ball rolling

Wouldn't it be best if you give it all of us, just to keep the bus factor high? (That does not entail any obligations). Maybe then it would also be better to archive the hpacheco/gloss-juicy repo and make this the main repo again? (I think it's always weird / confusing if the main repo is a fork)

**Comment by georgefst** [comment]
> Wouldn't it be best if you give it all of us, just to keep the bus factor high? (That does not entail any obligations). Maybe then it would also be better to archive the hpacheco/gloss-juicy repo and make this the main repo again? (I think it's always weird / confusing if the main repo is a fork)

Those both sounds like good ideas to me!

**Comment by hpacheco** [comment]
Sure, it makes sense to switch back to the main repo.
I can pass on Hackage rights to @georgefst and @ad-si, but I think that we should elect a front name to list as a *Maintainer* to avoid confusion, (even if it does not entail special obligations).

**Comment by alpmestan** [comment]
> Wouldn't it be best if you give it all of us, just to keep the bus factor high?

That'd be absolutely fine by me, I just wasn't sure everyone here was interested, hence the question. :)

**Comment by ad-si** [comment]
I just had another idea: How about I create a `haskell-contrib` organization (in the spirit of https://github.com/purescript-contrib/) and we transfer it there?
I'd also move https://github.com/ad-si/ulid there.

**Comment by hpacheco** [comment]
> I just had another idea: How about I create a `haskell-contrib` organization (in the spirit of https://github.com/purescript-contrib/) and we transfer it there?

you should probably propose that as a larger discussion in the haskell mailing list. there is already a [haskell github organization](https://github.com/haskell) but only for the core packages
I would say that it make most sense if it was integrated with Hackage

**Comment by alpmestan** [comment]
Yeah I think it's a bit overkill, especially since @hpacheco and I are still around to hand you all the rights needed to maintain this properly. If there was a gloss organization, perhaps that'd have been nice.

**Comment by georgefst** [comment]
In the end, @ad-si has forked Gloss itself and copied this package in as part of the project.

https://github.com/ad-si/Brillo/commit/932dd2c20e713591d4658a8b7cc0b00478d3e70b#diff-0a7c23dd5ffe7b83b6c48a025ea52d7598f5dd646c873e3d7dd47be7b42e4160R1

**Comment by ad-si** [comment]
> I just had another idea: How about I create a `haskell-contrib` organization (in the spirit of https://github.com/purescript-contrib/) and we transfer it there? I'd also move https://github.com/ad-si/ulid there.

Also, this happened in the meantime initiated by @jamesdbrock => https://github.com/haskell-github-trust

**Comment by jamesdbrock** [comment]
@alpmestan and @hpacheco I sent you ownership invitations to https://github.com/haskell-github-trust . 

If you want to transfer this package to the Trust, here are the instructions: https://github.com/haskell-github-trust#how-to-add-your-own-package-to-the-trust

## How to add your own package to the Trust

1. Transfer GitHub ownership of the package repository to this organization, [github.com/haskell-github-trust](https://github.com/haskell-github-trust).
2. Add the Hackage account [haskell_github_trust](https://hackage.haskell.org/user/haskell_github_trust) as a Hackage Maintainer for the package.

That’s it. We accept all packages, in any condition, with zero commitment or obligation.

**Comment by alpmestan** [comment]
@hpacheco has the most recent fork, but would be happy to transfer my repo if Hugo's not available within a couple of days, if we don't hear from him.

**Comment by jamesdbrock** [comment]
> @hpacheco has the most recent fork, but would be happy to transfer my repo if Hugo's not available within a couple of days, if we don't hear from him.

I suggest you transfer the original repo and then merge in the commits from the fork.

The forked repo is forever stigmatized by Github as “forked from.”
